app/models/user.rb

```diff
@@ -12,6 +13,7 @@
 #  article_count       :integer          default(0)
 #  revision_count      :integer          default(0)
 #  trained             :boolean          default(FALSE)
+# TODO: Use global_name instead
```

Comment:
`global_id` (which comes from the OAuth auth hash as `uid`, and is in fact the central_auth user id) is used to fetch the local user_id via replica (and local user_id is used other data from replica). Aside from that, we don't use it at all. Global username should always match regular username, as far as I know.

The main reason to use ids instead of usernames is to avoid encoding issues. Between ruby, php, mysql, and the wide range of allowed characters in mediawiki usernames, I've run into loads of bugs... which all went away when I switched to ids here: 40c2bbae28c4760d2435103d3ec93c2232e1d8b8